### PR TITLE
GH action to format and update use cases file

### PR DESCRIPTION
**Pull request by chusloj**
This PR introduces a new GH Action that will automatically pull the Tax-Calculator citations file (`use_cases.md`) with a CRON job that runs at 1 AM every day. Each time `use_cases.md` is updated, the ~~time &~~ date of the update will be recorded at the bottom of the file.

Other files changed:
- Conda environment – Added scripts necessary for pulling the original `.bib` file and formatting it.
- `bib_format.py` – Handles part of the formatting.

We'll need to add the Zotero API key to the repo's `Secrets` if we want to keep it private.

cc @MattHJensen

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -15,6 +15,9 @@ dependencies:
 - pylint
 - coverage
 - "paramtools>=0.14.2"
+- curl
+- pandoc
 - pip
 - pip:
     - jupyter-book
+    - pybtex
```
Comment (hdoupe) [inline environment.yml:23]: @chusloj Is it possible to only install these packages for the "Update Tax-Calculator Use Cases" action?

Comment (chusloj) [reply environment.yml:23]: @hdoupe Sure thing.

**Comment by chusloj** [comment]
@hdoupe Packages used just for this action have been removed from the environment file and `test_4package.py`.

**Comment by MattHJensen** [comment]
I'd rather the `Zotero -> PR to use_cases.md`  step be a responsibility of an OSPC RA rather than a responsibility of the Tax-Calculator codebase. 

Just musing, but to make things easier for you, would it be possible to set up this GH action on your fork of Tax-Calculator and then open a PR to Tax-Calculator when there are changes?

**Comment by chusloj** [comment]
@MattHJensen That would certainly be possible.

**Comment by chusloj** [comment]
A [new GH action](https://github.com/chusloj/Tax-Calculator/blob/citations_fork_action/.github/workflows/citations_pr.yml) has been setup on my `citations_fork_action` branch that check for updates on `use_cases.md` daily and will open a PR on my fork if there are any changes. I'll then open an upstream PR manually.

cc @MattHJensen

**Comment by chusloj** [comment]
Opening a PR upstream using GH actions is considerably more difficult than expected, so I'll just run CRON job from my local machine using the `gh` CLI tool.

Closing.

**Pull request closed without merging**